front/lib/tracking/customerio/server.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  LightWorkspaceType,
+  MembershipRoleType,
+  UserType,
+} from "@dust-tt/types";
+import * as _ from "lodash";
+
+import { subscriptionForWorkspace } from "@app/lib/auth";
+import { Workspace } from "@app/lib/models/workspace";
+import { MembershipResource } from "@app/lib/resources/membership_resource";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+
+const {
+  CUSTOMERIO_ENABLED,
+  CUSTOMERIO_SITE_ID = "",
+  CUSTOMERIO_API_KEY = "",
+} = process.env;
+
+const CUSTOMERIO_HOST = "https://track-eu.customer.io/api/v2";
+
+function isCustomerioEnabled() {
+  return CUSTOMERIO_ENABLED === "true";
+}
+
+export class CustomerioServerSideTracking {
+  static trackSignup({ user }: { user: UserType }) {
+    if (!isCustomerioEnabled()) {
+      return;
+    }
+    return CustomerioServerSideTracking._identifyUser({ user });
+  }
+
+  static async trackUserMemberships({ user }: { user: UserType }) {
+    if (!isCustomerioEnabled()) {
+      return;
+    }
+    const userMemberships = await MembershipResource.getLatestMemberships({
+      users: [user],
+    });
+
+    const workspaces = _.keyBy(
+      await Workspace.findAll({
+        where: {
+          id: userMemberships.map((m) => m.workspaceId),
+        },
+      }),
+      "id"
```

Comment:
According to my comment below, you can use the workspaces directly from the `UserTypeWithWorkspaces` from below.